src/System.Windows.Forms/src/System/Windows/Forms/TreeView.TreeViewAccessibleObject.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Drawing;
+using static Interop;
+using static System.Windows.Forms.TreeNode;
+
+namespace System.Windows.Forms
+{
+    public partial class TreeView
+    {
+        internal class TreeViewAccessibleObject : ControlAccessibleObject
+        {
+            private readonly TreeView _owningTreeView;
+
+            public TreeViewAccessibleObject(TreeView owningTreeView) : base(owningTreeView)
+            {
+                _owningTreeView = owningTreeView;
+            }
+
+            internal override UiaCore.IRawElementProviderFragmentRoot FragmentRoot => this;
+
+            internal override UiaCore.IRawElementProviderFragment? FragmentNavigate(UiaCore.NavigateDirection direction)
+                => direction switch
+                {
+                    UiaCore.NavigateDirection.FirstChild => GetChild(0),
+                    UiaCore.NavigateDirection.LastChild => GetChild(GetChildCount() - 1),
+                    _ => base.FragmentNavigate(direction),
+                };
+
+            public override AccessibleObject? GetChild(int index)
+                => index >= 0 && index < GetChildCount()
+                    ? _owningTreeView.Nodes[index].AccessibilityObject
+                    : null;
+
+            public override int GetChildCount() => _owningTreeView.Nodes.Count;
+
+            internal override int GetChildIndex(AccessibleObject? child)
+                => child is TreeNodeAccessibleObject node ? node.Index : -1;
+
+            internal override object? GetPropertyValue(UiaCore.UIA propertyID)
+                => propertyID switch
+                {
+                    UiaCore.UIA.ControlTypePropertyId => UiaCore.UIA.TreeControlTypeId,
+                    UiaCore.UIA.IsEnabledPropertyId => _owningTreeView.Enabled,
+                    UiaCore.UIA.IsKeyboardFocusablePropertyId => (State & AccessibleStates.Focusable) == AccessibleStates.Focusable,
+                    UiaCore.UIA.HasKeyboardFocusPropertyId => _owningTreeView.Nodes.Count == 0,
+                    _ => base.GetPropertyValue(propertyID)
+                };
+
+            public override AccessibleObject? HitTest(int x, int y)
+            {
+                if (!_owningTreeView.IsHandleCreated)
+                {
+                    return null;
+                }
+
+                Point p = _owningTreeView.PointToClient(new Point(x, y));
+                TreeNode node = _owningTreeView.GetNodeAt(p);
+
+                if (node is not null)
+                {
+                    return node.AccessibilityObject;
+                }
+
+                if (Bounds.Contains(x, y))
+                {
+                    return this;
+                }
+
+                return null;
+            }
+
```

Comment:
Sorry for the late, but I've found how to fix #6558 - you need to provide an implementation for `ElementProviderFromPoint` method. It is usually implemented to call `HitTest`:
```suggestion

            internal override IRawElementProviderFragment? ElementProviderFromPoint(double x, double y)
                => HitTest((int)x, (int)y) ?? base.ElementProviderFromPoint(x, y);

```

Reply:
Just in time, before closing it :) Great investigation! Thanks, it works

Reply:
It doesn't affect the above described problem with focus on selected node first time, I've checked it